Mathlib/RingTheory/Ideal/Norm/RelNorm.lean

```diff
@@ -199,6 +195,85 @@ theorem spanNorm_mul (I J : Ideal S) : spanNorm R (I * J) = spanNorm R I * spanN
     span_singleton_mul_span_singleton, spanNorm_singleton, spanNorm_singleton,
     spanNorm_singleton, span_singleton_mul_span_singleton, map_mul]
 
+section spanNorm_spanNorm
+
+variable (T : Type*) [CommRing T] [IsDomain T] [IsIntegrallyClosed T] [Algebra R T] [Algebra T S]
+  [Module.Finite R T] [Module.Finite T S] [NoZeroSMulDivisors R T] [NoZeroSMulDivisors T S]
+  [IsScalarTower R T S] [Algebra.IsSeparable (FractionRing R) (FractionRing T)]
+  [Algebra.IsSeparable (FractionRing T) (FractionRing S)]
+
+open _root_.Algebra
+
+theorem le_spanNorm_spanNorm (I : Ideal S) : spanNorm R I ≤ spanNorm R (spanNorm T I) := by
+  simp_rw [spanNorm, map]
+  refine span_mono ?_
+  rintro _ ⟨x, hx, rfl⟩
+  exact ⟨intNorm T S x, subset_span <| Set.mem_image_of_mem _ hx, by rw [intNorm_intNorm]⟩
+
+/--
+This condition `eq_bot_or_top` is equivalent to being a field. However,
+`Ideal.spanNorm_spanNorm_of_field` would be harder to apply since we'd need to upgrade
+a `CommRing R` instance to a `Field R` instance.
+-/
+theorem spanNorm_spanNorm_of_bot_or_top (eq_bot_or_top : ∀ I : Ideal R, I = ⊥ ∨ I = ⊤)
+    (I : Ideal S) : spanNorm R (spanNorm T I) = spanNorm R I := by
+  obtain h | h := eq_bot_or_top (spanNorm R I)
+  · rw [h, spanNorm_eq_bot_iff, spanNorm_eq_bot_iff, spanNorm_eq_bot_iff.mp h]
+  · exact h ▸ (eq_top_iff_one _).mpr <| le_spanNorm_spanNorm R T I <| (eq_top_iff_one _).mp h
+
+theorem spanNorm_spanNorm [IsDedekindDomain R] [IsDedekindDomain T] [IsDedekindDomain S]
+    (I : Ideal S) : spanNorm R (spanNorm T I) = spanNorm R I := by
+  refine eq_of_localization_maximal fun P hP ↦ ?_
+  by_cases hP : P = ⊥
+  · subst hP
+    rw [spanNorm_spanNorm_of_bot_or_top]
+    exact fun I ↦ or_iff_not_imp_right.mpr fun hI ↦ (hP.eq_of_le hI bot_le).symm
+  let Mₜ := Algebra.algebraMapSubmonoid T P.primeCompl
```

Comment:
I've added `Mathlib.RingTheory.DedekindDomain.Instances` to avoid long series of `have` in proofs about Dedekind domains. Maybe some of those can already be removed, bu otherwise it's a good idea to add the general instance.

Reply:
The results here are not covered by the content of ` Mathlib.RingTheory.DedekindDomain.Instances`. Do you think I should add these instances the file? Note that, in #27706, I prove results about extensions of localizations but that is in the case where `p.primeCompl` is used to localized both rings.

Reply:
Yes, if you don't mind splitting would be a good thing (ask my review!)